probe/endpoint/ebpf.go

```diff
@@ -0,0 +1,226 @@
+	hasDied() bool
+	run()
+	walkFlows(f func(ebpfConnection))
+	initialize()
+	isInitialized() bool
+}
+
+// nilTracker is a tracker that does nothing, and it implements the eventTracker interface.
+// It is returned when the ebpfEnabled flag is false.
+type nilTracker struct{}
+
+func (n nilTracker) handleFlow(_ string, _ fourTuple, _ int, _ string) {}
+func (n nilTracker) hasDied() bool                                     { return true }
+func (n nilTracker) run()                                              {}
+func (n nilTracker) walkFlows(f func(ebpfConnection))                  {}
+func (n nilTracker) initialize()                                       {}
+func (n nilTracker) isInitialized() bool                               { return false }
+
+// EbpfTracker contains the list of eBPF events, and the eBPF script's command
+type EbpfTracker struct {
+	sync.Mutex
+	cmd *exec.Cmd
+
+	initialized bool
+	dead        bool
+
+	activeFlows   map[string]ebpfConnection
+	bufferedFlows []ebpfConnection
+}
+
+func newEbpfTracker(ebpfEnabled bool) eventTracker {
+	if !ebpfEnabled {
+		return &nilTracker{}
+	}
+	cmd := exec.Command(TCPV4TracerLocation)
+	env := os.Environ()
+	cmd.Env = append(env, "PYTHONUNBUFFERED=1")
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		log.Errorf("EbpfTracker error: %v", err)
+		return nil
+	}
+	go logPipe("EbpfTracker stderr:", stderr)
+
+	tracker := &EbpfTracker{
+		cmd:         cmd,
+		activeFlows: map[string]ebpfConnection{},
+	}
+	log.Info("EbpfTracker started")
+	go tracker.run()
+	return tracker
+}
+
+func (t *EbpfTracker) handleFlow(eventType string, tuple fourTuple, pid int, networkNamespace string) {
+	t.Lock()
+	defer t.Unlock()
+
+	switch eventType {
+	case "connect":
+		conn := ebpfConnection{
+			incoming:         false,
+			tuple:            tuple,
+			pid:              pid,
+			networkNamespace: networkNamespace,
+		}
+		t.activeFlows[tuple.String()] = conn
+	case "accept":
+		conn := ebpfConnection{
+			incoming:         true,
+			tuple:            tuple,
+			pid:              pid,
+			networkNamespace: networkNamespace,
+		}
+		t.activeFlows[tuple.String()] = conn
+	case "close":
+		if deadConn, ok := t.activeFlows[tuple.String()]; ok {
+			delete(t.activeFlows, tuple.String())
+			t.bufferedFlows = append(t.bufferedFlows, deadConn)
+		} else {
+			log.Errorf("EbpfTracker error: unmatched close event: %s pid=%d netns=%s", tuple.String(), pid, networkNamespace)
+		}
+	}
+
+}
+
+func (t *EbpfTracker) run() {
+	stdout, err := t.cmd.StdoutPipe()
+	if err != nil {
+		log.Errorf("EbpfTracker error: %v", err)
+		return
+	}
+
+	if err := t.cmd.Start(); err != nil {
+		log.Errorf("EbpfTracker error: %v", err)
+		return
+	}
+
+	defer func() {
+		if err := t.cmd.Wait(); err != nil {
+			log.Errorf("EbpfTracker error: %v", err)
+		}
+
+		t.Lock()
+		t.dead = true
+		t.Unlock()
+	}()
+
+	reader := bufio.NewReader(stdout)
+	// skip fist line
```

Comment:
s/fist/first

either explain why you skip the first line or just remove that, add a --no-header option to the python script so that it does not print the header.